bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/link/ItemChannelLinkResource.java

```diff
@@ -80,9 +82,13 @@ public Response isAutomatic() {
     @Path("/{itemName}/{channelUID}")
     @ApiOperation(value = "Links item to a channel.")
     @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
-            @ApiResponse(code = 400, message = "Item already linked to the channel.") })
+            @ApiResponse(code = 400, message = "Item already linked to the channel."),
+            @ApiResponse(code = 405, message = "Item name invalid.") })
     public Response link(@PathParam("itemName") @ApiParam(value = "itemName") String itemName,
             @PathParam("channelUID") @ApiParam(value = "channelUID") String channelUid) {
+        if (!itemRegistry.isValidItemName(itemName)) {
```

Comment:
I do not really like to introduce a dependency to the itemRegistry for this method call.
The endpoint is about creating a link. If the content is not valid, either the constructor of the ItemChannelLink()  should complain or at least the ChannelLinkRegistry should not accept the link in add().